pyccel/ast/basic.py

```diff
@@ -63,12 +63,12 @@ def __init__(self):
 
             if isinstance(c, tuple):
                 for ci in c:
-                    if not self.ignore(ci):
+                    if not Basic._ignore(self, ci):
                         ci.set_current_user_node(self)
             else:
                 c.set_current_user_node(self)
 
-    def ignore(self, c):
+    def _ignore(self, c):
         """ Indicates if a node should be ignored when recursing
         """
         return c is None or isinstance(c, self._ignored_types)
```

Comment:
Given the fact that `_ignored_types` is a class attribute, the behavior of this function does not depend on the `self` object. Hence I think it should be a class method (decorated with `@classmethod`).

Reply:
Fixed [6de7d35](https://github.com/pyccel/pyccel/pull/1113/commits/6de7d35e5ab13d6b0e04bd64cd7f0507cf511d9d)